From f00c2438d8aacde8d53fd4d428bd05f42e485e29 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Fri, 25 Apr 2025 15:10:56 +0200
Subject: [PATCH] arm64: dts: st: fix address conflict between SCMI SHM and
 IPCC SHM

Currently, the server must be addressed by M33 (secure or non secure)
for some SCMI requests. As symmetric approach, we will use the same
SRAM1 shared memory as used in M33-TD.

It needs 4kB for each mailbox:
 - scmi_cid2_ns for M33_NS to A35S
 - scmi_cid2_s for M33_S to A35_S

Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I5b0c5cc85dec4598c74a2e9156b1a73f8dd2be1c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/454216
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 .../dts/st/stm32mp215f-dk-ca35tdcid-resmem.dtsi    | 14 ++++++++++++--
 .../dts/st/stm32mp235f-dk-ca35tdcid-resmem.dtsi    | 14 ++++++++++++--
 .../dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi    | 14 ++++++++++++--
 .../dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi   | 14 ++++++++++++--
 4 files changed, 48 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp215f-dk-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp215f-dk-ca35tdcid-resmem.dtsi
index 41645838e931..37aa458f644e 100644
--- a/arch/arm64/boot/dts/st/stm32mp215f-dk-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk-ca35tdcid-resmem.dtsi
@@ -42,8 +42,18 @@ bsec_mirror: bsec-mirror@a060000 {
 			no-map;
 		};
 
-		cm33_sram1: cm33-sram1@a061000 {
-			reg = <0x0 0xa061000 0x0 0x8000>;
+		scmi_cid2_s: scmi-cid2-s@a061000 {
+			reg = <0x0 0xa061000 0x0 0x1000>;
+			no-map;
+		};
+
+		scmi_cid2_ns: scmi-cid2-ns@a062000 {
+			reg = <0x0 0xa062000 0x0 0x1000>;
+			no-map;
+		};
+
+		cm33_sram1: cm33-sram1@a063000 {
+			reg = <0x0 0xa063000 0x0 0xd000>;
 			no-map;
 		};
 
diff --git a/arch/arm64/boot/dts/st/stm32mp235f-dk-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp235f-dk-ca35tdcid-resmem.dtsi
index 059fb15297a1..5d669b01cba3 100644
--- a/arch/arm64/boot/dts/st/stm32mp235f-dk-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp235f-dk-ca35tdcid-resmem.dtsi
@@ -42,8 +42,18 @@ bsec_mirror: bsec-mirror@a040000 {
 			no-map;
 		};
 
-		cm33_sram1: cm33-sram1@a041000 {
-			reg = <0x0 0xa041000 0x0 0x1f000>;
+		scmi_cid2_s: scmi-cid2-s@a041000 {
+			reg = <0x0 0xa041000 0x0 0x1000>;
+			no-map;
+		};
+
+		scmi_cid2_ns: scmi-cid2-ns@a042000 {
+			reg = <0x0 0xa042000 0x0 0x1000>;
+			no-map;
+		};
+
+		cm33_sram1: cm33-sram1@a043000 {
+			reg = <0x0 0xa043000 0x0 0x1d000>;
 			no-map;
 		};
 
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
index 243df32b685d..2931ad2b4594 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
@@ -42,8 +42,18 @@ bsec_mirror: bsec-mirror@a040000 {
 			no-map;
 		};
 
-		cm33_sram1: cm33-sram1@a041000 {
-			reg = <0x0 0xa041000 0x0 0x1f000>;
+		scmi_cid2_s: scmi-cid2-s@a041000 {
+			reg = <0x0 0xa041000 0x0 0x1000>;
+			no-map;
+		};
+
+		scmi_cid2_ns: scmi-cid2-ns@a042000 {
+			reg = <0x0 0xa042000 0x0 0x1000>;
+			no-map;
+		};
+
+		cm33_sram1: cm33-sram1@a043000 {
+			reg = <0x0 0xa043000 0x0 0x1d000>;
 			no-map;
 		};
 
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
index 58266b81c7cf..265338977023 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem.dtsi
@@ -42,8 +42,18 @@ bsec_mirror: bsec-mirror@a040000 {
 			no-map;
 		};
 
-		cm33_sram1: cm33-sram1@a041000 {
-			reg = <0x0 0xa041000 0x0 0x1f000>;
+		scmi_cid2_s: scmi-cid2-s@a041000 {
+			reg = <0x0 0xa041000 0x0 0x1000>;
+			no-map;
+		};
+
+		scmi_cid2_ns: scmi-cid2-ns@a042000 {
+			reg = <0x0 0xa042000 0x0 0x1000>;
+			no-map;
+		};
+
+		cm33_sram1: cm33-sram1@a043000 {
+			reg = <0x0 0xa043000 0x0 0x1d000>;
 			no-map;
 		};
 
-- 
2.50.1

